src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+// import edu.wpi.first.wpilibj2.*;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import edu.wpi.first.wpilibj.DriverStation;
+
+/** Add your docs here. */
+// Put methods for controlling this subsystem
+// here. Call these from Commands.
+public class BatterySubsystem extends SubsystemBase {
+
+  private ShuffleboardTab btTab;
+  private NetworkTableEntry sbVoltage;
+  private NetworkTableEntry sbInCurrent;
+  private NetworkTableEntry sbTimer;
+  private NetworkTableEntry sbTimerChange;
+  private NetworkTableEntry sbTimerHighCurrent;
+  private DriverStation driverStation;
+  private Timer timer = new Timer();
+  private Timer currentTimer = new Timer();
+
+  private void init() {
+    //timer.reset(); Only uncomment this if you want the timer to reset each time the robot redeploys code
+    btTab = Shuffleboard.getTab("Battery");
+    sbVoltage = btTab.add("Battery Voltage", 0).withSize(2, 2).withPosition(0, 0).getEntry();
+    sbInCurrent = btTab.add("Input Current", 0).withSize(2, 2).withPosition(2, 0).getEntry();
+    sbTimer = btTab.add("Timer", 0).withSize(2, 2).withPosition(0, 2).getEntry();
+    sbTimerHighCurrent = btTab.add("High Current Timer", 0).withSize(2, 2).withPosition(0, 4).getEntry();
+    sbTimerChange = btTab.add("Change Timer", 0).withSize(2, 2).withPosition(4, 0).getEntry();
+    timer.start();
+    currentTimer.start();
+  }
+
+  public BatterySubsystem() {
+    init();
```

Comment:
Is there a reason we need a separate private method for the constructor?

Reply:
fixed to just initialize when called